src/MongoDB.Driver.Core/Core/NativeLibraryLoader/ILibraryLocator.cs

```diff
@@ -0,0 +1,61 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace MongoDB.Driver.Core.NativeLibraryLoader
+{
+    internal interface ILibraryLocator
+    {
+        string GetLibraryAbsolutePath(SupportedPlatform currentPlatform);
+    }
+
+    internal abstract class LibraryRelativeLocatorBase : ILibraryLocator
+    {
+        public virtual Assembly Assembly => typeof(LibraryRelativeLocatorBase).GetTypeInfo().Assembly;
```

Comment:
I would prefer to see a method than a property since a small amount of work is being done. Also, we can factor apart the notions of "base assembly" and "base path" where the default "base path" is computed from the default "base assembly". Subclasses can either override `GetLibraryBaseAssembly` or `GetLibraryBasePath`:

    // public methods
    public string GetLibraryAbsolutePath(SupportedPlatform currentPlatform)
    {
        var relativePath = GetLibraryRelativePath(currentPlatform);
        return GetAbsolutePath(relativePath);
    }

    public virtual Assembly GetLibraryBaseAssembly()
    {
        return typeof(RelativeLibraryLocatorBase).GetTypeInfo().Assembly;
    }

    public virtual string GetLibraryBasePath()
    {
        var assembly = GetLibraryBaseAssembly();
        var codeBase = assembly.CodeBase;
        var uri = new Uri(codeBase);
        var absolutePath = uri.AbsolutePath;
        return Path.GetDirectoryName(absolutePath);
    }

    public abstract string GetLibraryRelativePath(SupportedPlatform currentPlatform);

Reply:
done